nativepython/type_wrappers/string_wrapper.py

```diff
@@ -137,3 +141,63 @@ def constant(self, context, s):
                 ).cast(self.layoutType)
             )
         )
+
+    def convert_attribute(self, context, instance, attr):
+        if attr in ("lower", "find"):
+            return instance.changeType(BoundCompiledMethodWrapper(self, attr))
+
+        return super().convert_attribute(context, instance, attr)
+
+    def convert_method_call(self, context, instance, methodname, args, kwargs):
+        if kwargs:
+            return super().convert_method_call(context, instance, methodname, args, kwargs)
+
+        if methodname == "lower":
+            if len(args) != 0:
+                    return
```

Comment:
Here we should fall through to the `super().convert_method_call(...)`. That generates code that throws an exception at runtime, which we want to do. As written, if code calls `lower` with more than one argument compilation will fail.

We should also be testing this behavior.